### DISCUSSION TITLE
Use TOPP-RA as submodule in a CMake C++ project.

### DISCUSSION THREAD

**Discussion by maxpla3** [hungpham2511/toppra#222] [Q&A] [answered]
Is it possible to use this repository as a submodule in a C++ project? I added this repo via `git submodule add` and included it into my cmake configuration via `find_packages(toppra)`. But I'm getting the following error:
```
1> [CMake]   Could not find a package configuration file provided by "toppra" with any
1> [CMake]   of the following names:
1> [CMake] 
1> [CMake]     toppraConfig.cmake
1> [CMake]     toppra-config.cmake
```

**Comment by maxpla3** [comment, marked as answer]
Solution:
First, add toppra as a git submodule in your project.
Then, add following lines to the CMakeLists.txt of your project (assuming your executable is called "main"):

```
target_link_libraries ("main" toppra)

target_include_directories ("main" PUBLIC ${PROJECT_SOURCE_DIR}/toppra/cpp/src)

set (BUILD_TESTS OFF CACHE BOOL "Build Gtests")
set (BUILD_SHARED_LIBS OFF CACHE BOOL "Build shared libraries")
set (PYTHON_BINDINGS OFF CACHE BOOL "Build bindings for Python")
add_subdirectory (toppra/cpp)
```

Set BUILD_SHARED_LIBS to ON if you'd like to use shared libs.

**Reply by hungpham2511** [reply]
Thanks!
